Makefile

```diff
@@ -1,11 +1,15 @@
-PREFIX=/usr/
-DESTDIR=/
-BINARY_NAME=apx
+PREFIX := /usr
+DESTDIR := /
+BINARY_NAME := apx
+
+GO := go
 
 all: build
 
-build:
-	CGO_ENABLED=0 GOOS=linux GOARCH=amd64 go build -a -tags netgo -ldflags '-w -extldflags "-static"' -o ${BINARY_NAME}
+build: ${BINARY_NAME}
+
+${BINARY_NAME}:
+	CGO_ENABLED=0 GOOS=linux GOARCH=amd64 ${GO} build -a -tags netgo -ldflags '-w -extldflags "-static"' -o $@
```

Comment:
why exactly is this done? i dont see any benefit

Reply:
It's because of how Makefiles work - files need to be specified as targets to avoid rebuilding them every time.
Before, `install` called `build` even when the binary was already present, so `make && make install` resulted in two builds. This fixes that

Reply:
right yeah, though that also was my intention, since it'd be annoying to run a clean before building while developing apx
maybe remove the dependency on the build target for install instead?

Reply:
Well, `install` depending on `build` is standard practice from what I've seen, and it means that the user can simply run `make install` for a build & install.
But I don't feel strongly on that, one way or another

Reply:
In my opinion it's not that bad of a deal to have it compile again on install, it's not like apx is a massive project that takes ages to build

Reply:
It's a bigger deal on low-powered CI runners. It's generally annoying for packaging where the build and install steps are separate, since the Makefile needs to be patched to avoid a double build

Reply:
> right yeah, though that also was my intention, since it'd be annoying to run a clean before building while developing apx
`make clean build` is 6 more characters

Reply:
maybe add an `all` target which does a clean first

Reply:
Done